gems/aws-sdk-core/lib/aws-sdk-core/middleware/checksums.rb

```diff
@@ -57,13 +57,20 @@ def call(input, context)
 
         output = @app.call(input, context)
         @response_validation_body&.validate!
+        if context.metadata[:http_checksum][:validated]
+          output.metadata[:http_checksum] = {
+            validated: context.metadata[:http_checksum][:validated]
+          }
+        end
         output
       end
 
       private
 
       def request_checksum(input, context)
-        checksum_algorithm = input[@request_algorithm_member]
+        if @request_algorithm_member
```

Comment:
I saw a case where this was nil (you just render stack.use(Checksum) with no options). When does this happen?

Reply:
This happens when there is not request validation (only response, as in put_object), which should render something like:
```
      stack.use(AWS::SDK::Core::Middleware::Checksum,
        response_algorithms: ['CRC32', 'CRC32C', 'SHA256', 'SHA1'],
        request_validation_mode_member: :checksum_mode
      )
```

Where did you find a stack.use(Checksum) with no options - that shouldn't happen I think - I think theres validation on the checksum trait that it must have at least one of request/response)